packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -41,15 +41,13 @@ type AuthDataKeys =
 type AuthDataMap = Record<
```

Comment:
Not relevant to this PR, but I was curious if there was a way to use `TokenResponse` so that we don't need to manually list the keys. Turns out there is!

```ts
// Helpers
type RemoveNeverValues<T> = {
    // Key remapping, fancy stuff! https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#key-remapping-in-mapped-types
    [K in keyof T as T[K] extends never ? never: K]: T[K]
}
type StringIndexNever<T> = {
    [K in keyof T]: string extends K ? never : T[K]
}
type KnownKeys<T> = keyof RemoveNeverValues<StringIndexNever<T>>

// Implementation
type TokenResponseKeys = KnownKeys<ShopperLoginTypes.TokenResponse>
type AuthDataKeys =
    | Exclude<TokenResponseKeys, 'refresh_token'>
    | 'idp_access_token'
    | 'refresh_token_guest'
    | 'refresh_token_registered'
    | 'site_id'
```

Reply:
@wjhsf This is nice 👌, this either close to (if not exactly) what I was mentioning in the typescript chat we had with @kevinxh last week

Reply:
Looks promising to my TS n00b eyes. Maybe you can contribute this is a separate PR? @wjhsf